src/geometry/WorkingGeometry.ts

```diff
@@ -303,16 +328,20 @@ export class WorkingGeometry implements Bakeable {
             child.combine();
         });
         let vertexCount = this.vertices.length;
+        let normalCount = this.normals.length;
         for (const child of this.mergedObjects) {
             this.vertices = this.vertices.concat(child.vertices);
             this.normals = this.normals.concat(child.normals);
             child.faces.forEach((face: Face) => {
-                const newIndices = face.indices.map((i: number) => i + vertexCount);
-                face.indices = newIndices;
+                const newVertexIndices = face.vertexIndices.map((i: number) => i + vertexCount);
+                const newNormalIndices = face.normalIndices.map((i: number) => i + normalCount);
+                face.vertexIndices = newVertexIndices;
```

Comment:
nit: I personally think I would find this easier to read if we inline `newVertexIndices` (just put the map expression on the right side). Same with `newNormalIndices`, but that's just a personal preference.